wallet-new/src/Cardano/Wallet/API/V1/Errors.hs

```diff
@@ -164,6 +165,23 @@ toHttpStatus :: WalletError -> HTTP.Status
 toHttpStatus err = HTTP.Status (errHTTPCode $ toServantError err)
                                (encodeUtf8 $ describe err)
 
+transactionErrorToWalletError :: TxError.TxError -> WalletError
+transactionErrorToWalletError = \case
+    TxError.NotEnoughMoney coin ->
+        NotEnoughMoney (fromIntegral (Core.getCoin coin))
+    TxError.NotEnoughAllowedMoney coin ->
+        NotEnoughMoney (fromIntegral (Core.getCoin coin))
+    TxError.FailedToStabilize ->
+        UnknownError "Transaction process failed to stabilize."
+    TxError.OutputIsRedeem addr ->
+        OutputIsRedeem (show addr)
```

Comment:
The type of `OutputIsRedeem` is `OutputIsRedeem !Address`, which means we are being passed a nice Cardano's `Address` but we are now flattening it into a raw string.

This wouldn't be in principle a problem here, if not that showing it via the `Show` instance will (AFAIK) bump the raw internal structure as an output -- what you probably want is to use the `Buildable` instance, which will render this as a nice base58 string 😉 

I'm wondering if indeed our `WalletError.OutputIsRedeem` is a poor constructor -- what about:

1. Changing its type to take a `V1 Address` (basically almost _exactly_ as `TxError.OutputIsRedeem`)?

If I'm not mistaken, this will also ensure that rendering such type constructor as a `JSON` will trigger `ToJSON (V1 Address)` which should correctly render it, as `ToJSON (V1 Address)` is defined as:

```haskell
instance ToJSON (V1 Core.Address) where
    toJSON (V1 c) = String $ sformat addressF c
```

Reply:
Good call :)